atomica/model.py

```diff
@@ -2248,6 +2322,23 @@ def build(self, parset):
                 obj.preallocate(self.t, self.dt)
             pop.initialize_compartments(parset, self.framework, self.t[0])
 
+        #More finally, set up acceptance criteria for those which should be checked at runtime (variables that have ._is_dynamic = True), and those which can only be checked after conclusion
+        if self.acceptance_criteria:
```

Comment:
Maybe we should be more generic about this in Atomica e.g., support a generic callable and defer the specific validation method to the user